java-lib/src/main/java/com/wavefront/metrics/JsonMetricsGenerator.java

```diff
@@ -259,7 +312,7 @@ public static void writeRegularMetrics(Processor processor, JsonGenerator json,
           e.printStackTrace();
         }
         // need to close the object as well.
-        if (subEntry.getKey() instanceof TaggedMetricName) {
+        if (subEntry.getKey() instanceof TaggedMetricName || pointTags != null) {
```

Comment:
please use a variable and share it with line 274, these are easy to miss if one changes one place and not the other